source/material_model/visco_plastic.cc

```diff
@@ -815,6 +929,107 @@ namespace aspect
                                    "standard viscosity rescaling method (concretely, values $n_y>20$ "
                                    "are large enough)."
                                    "\n\n "
+                                   "The visco-plastic rheology described above may also be modified to include "
+                                   "viscoelastic deformation, thus producing a viscoelastic plastic constitutive "
+                                   "relationship. "
+                                   "\n\n "
+                                   "The viscoelastic rheology behavior takes into account the elastic shear "
```

Comment:
Independent of this PR, but can I get you to write a section in the manual with these formulas? Something similar to section 2.12 or 2.13. This might be a better location for references and this lengthy description.

Reply:
Yes, will do. I've opened an issue (#3365) and assigned myself.